src/Compilers/Core/Portable/StrongName/CryptoBlobParser.cs

```diff
@@ -165,106 +144,55 @@ private struct BlobHeader
             public UInt32 AlgId;    // Algorithm ID. Must be one of ALG_ID specified in wincrypto.h
         }
 
+        private const int BlobHeaderSize = 8;
+
         // RSAPUBKEY struct from wincrypt.h
-        [StructLayout(LayoutKind.Sequential)]
         private struct RsaPubKey
         {
             public UInt32 Magic; // Indicates RSA1 or RSA2
             public UInt32 BitLen; // Number of bits in the modulus. Must be multiple of 8.
             public UInt32 PubExp; // The public exponent
         }
+
+        private const int RsaPubKeySize = 12;
+
         private const UInt32 RSA1 = 0x31415352;
         private const UInt32 RSA2 = 0x32415352;
 
         // In wincrypt.h both public and private key blobs start with a
         // PUBLICKEYSTRUC and RSAPUBKEY and then start the key data
-        private unsafe static readonly int s_offsetToKeyData = sizeof(BlobHeader) + sizeof(RsaPubKey);
+        private const int s_offsetToKeyData = BlobHeaderSize + RsaPubKeySize;
 
-        private static ImmutableArray<byte> CreateSnPublicKeyBlob(BlobHeader header, RsaPubKey rsa, byte[] pubKeyData)
+        private static ImmutableArray<byte> CreateSnPublicKeyBlob(ref BlobHeader header, ref RsaPubKey rsa, byte[] pubKeyData)
```

Comment:
>ref BlobHeader header, ref RsaPubKey rsa [](start = 66, length = 40)

Can we get rid of the structs and pass in all the necessary values as parameters? I think it would clean up the callers.